docs/configuration.asciidoc

```diff
@@ -15,20 +13,319 @@ server_url: 'http://localhost:8200'
+With the above settings, you can modify how many lines of source code is collected.
+
+We differ between errors and spans, as well as library frames and app frames.
+
+WARNING: Especially for spans, collecting source code can have a large impact on
+storage use in your Elasticsearch cluster.
+
+[float]
+[[config-max-queue-size]]
+==== `max_queue_size`
+
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_MAX_QUEUE_SIZE` | `max_queue_size` | `500`
+|============
+
+Maximum queue length of transactions before sending transactions to the APM server.
+A lower value will increase the load on your APM server,
+while a higher value can increase the memory pressure of your app.
+A higher value also impacts the time until transactions are indexed and searchable
+in Elasticsearch.
+
+This setting is useful to limit memory consumption if you experience a sudden spike
+of traffic.
+
+[float]
+[[config-flush-interval]]
+==== `flush_interval`
+
+|============
+| Environment                  | `Config` key     | Default
+| `ELASTIC_APM_FLUSH_INTERVAL` | `flush_interval` | `10`
+|============
+
+Interval with which transactions should be sent to the APM server, in seconds.
+A lower value will increase the load on your APM server,
+while a higher value can increase the memory pressure on your app.
+A higher value also impacts the time until transactions are indexed and searchable
+in Elasticsearch.
+
+[float]
+[[config-transaction-sample-rate]]
+==== `transaction_sample_rate`
+
+|============
+| Environment                           | `Config` key              | Default
+| `ELASTIC_APM_TRANSACTION_SAMPLE_RATE` | `transaction_sample_rate` | `1.0`
+|============
+
+By default, the agent will sample every transaction (e.g. request to your service).
+To reduce overhead and storage requirements, you can set the sample rate to a value
+between `0.0` and `1.0`.
+We still record overall time and the result for unsampled transactions, but no
+context information, tags, or spans.
+
+[float]
+[[config-transaction-max-spans]]
+==== `transaction_max_spans`
+
+|============
+| Environment                         | `Config` key            | Default
+| `ELASTIC_APM_TRANSACTION_MAX_SPANS` | `transaction_max_spans` | `500`
+|============
+
+Limits the amount of spans that are recorded per transaction.
+This is helpful in cases where a transaction creates a very high amount of spans
+(e.g. thousands of SQL queries).
+Setting an upper limit will prevent overloading the agent and the APM server with
+too much work for such edge cases.
+
+[float]
+[[config-verify-server-cert]]
+==== `verify_server_cert`
+|============
+| Environment                       | `Config` key         | Default
+| `ELASTIC_APM_VERIFY_SERVER_CERT`  | `verify_server_cert` | `true`
+|============
+
+By default, the agent verifies the SSL certificate if you use an HTTPS connection to
+the APM server.
+Verification can be disabled by changing this setting to `false`.
```

Comment:
I'd add a  note, that this is generally not recommended (but might be helpful for testing)